### PR TITLE
Collection delete

### PR DESCRIPTION
**Pull request by jerabekjiri**
Issue: [AAH-715](https://issues.redhat.com/browse/AAH-715)

This PR includes both deleting collection version and entire collection. 

- API endpoints for deleting collection 
     - Deleting collection version (on master)
     - Deleting entire collection (on master)

- Reused `ConfirmModal` for deleting collections modal
- Deleting flows:
    - **Deleting entire collection**: After successful deletion, redirect to collection's namespace detail and show success alert 
    - **Deleting last collection version**: same as above, because collection doesn't exist anymore, we redirect the user to namespace detail   
    - **Deleting version collection**: After deleting specific version, reload page, show latest version and display success alert
    
- Unable to delete collection when collections depend on this collection
- Updated: always show `Delete version ___`
Screenshot from 2021-09-21 13-07-39
- Modal window for deleting entire collection
Screenshot from 2021-09-19 15-46-15
- Modal window for deleting specific collection version
Screenshot from 2021-09-19 15-50-29

- Modal window for deleting collection version that is last in the collection, this deletes entire collection, after successful collection deletion we redirect user to the collection's namespace
Screenshot from 2021-09-19 15-46-02

- Alert on successfully deleting collection version
Screenshot from 2021-09-19 15-50-41
- Alert on successfully deleting entire collection 
Screenshot from 2021-09-19 15-51-01

- Error alert on deleting collection version

Screenshot from 2021-09-21 13-08-03

### CODE REVIEW

**Comment by ZitaNemeckova** [comment]
I tried to delete a collection that comes from `rh-certified` and it failed without showing any error message. 

Screenshot 2021-09-21 at 14 25 53

```
DELETE http://localhost:8002/api/automation-hub/v3/collections/arista/eos/versions/3.0.0/ 404 (Not Found)
collection-header.tsx:533 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'map')
    at collection-header.tsx:533
```

I have a same collection in `published` and `rh-certified` that is a dependency of another collection in `rh-certified`. I cannot delete it when in `published` repo because of dependency in. `rh-certified` repo. And I cannot delete anything in `rh-certified`.


Is a user allowed to delete a collection other than in `published` repo?

File: src/api/collection.ts [hunk 1]
```diff
@@ -165,6 +165,27 @@ export class API extends HubAPI {
         .catch((err) => reject(err));
     });
   }
+
+  deleteCollectionVersion(collection) {
+    return this.http.delete(
+      `v3/collections/${collection.namespace.name}/${collection.name}/versions/${collection.latest_version.version}/`,
```
Comment (newswangerd) [inline src/api/collection.ts:171]: This has to be prefixed with the repository that the content is in. The endpoint is `/content/<repository>/v3...`

Comment (jerabekjiri) [reply src/api/collection.ts]: It's working now, thanks :)

**Comment by ZitaNemeckova** [comment]
@jerabekjiri This needs a rebase. And I still see error in https://github.com/ansible/ansible-hub-ui/pull/947#issuecomment-923937651 butt that can be resolved in followup PR.

**Comment by jerabekjiri** [comment]
- rebased
- improved dependencies error alert
Screenshot from 2021-09-24 13-33-42

@ZitaNemeckova I thought that would be fixed by resolving the issue with the wrong API path. I was trying to reproduce it, but was unable to upload two same dependencies, also from talking to Awcrosby: `CollectionVersion could exist in more than one Repository, and when a delete occurs it happens across entire server so it is deleted from all Repository` it shouldn't a problem then? I'll look into it more :)

**Review by ZitaNemeckova — APPROVED** [review]
LGTM :+1:

**Pull request merged**